src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -3864,6 +3928,26 @@ private void CopyFileInfoItem(FileInfo file, string destinationPath, bool force,
 
                             FileInfo result = new FileInfo(destinationPath);
                             WriteItemObject(result, destinationPath, false);
+
+                            if (_totalFiles > 0)
+                            {
+                                _copiedFiles++;
+                                _copiedBytes += file.Length;
+                                double speed = (double)(_copiedBytes / 1024 / 1024) / _copyStopwatch.Elapsed.TotalSeconds;
+                                var progress = new ProgressRecord(
+                                    COPY_FILE_ACTIVITY_ID,
+                                    StringUtil.Format(FileSystemProviderStrings.CopyingLocalFileActivity, _copiedFiles, _totalFiles),
+                                    StringUtil.Format(FileSystemProviderStrings.CopyingLocalBytesStatus, Utils.DisplayHumanReadableFileSize(_copiedBytes), Utils.DisplayHumanReadableFileSize(_totalBytes), speed)
+                                );
+                                var percentComplete = (int)((_copiedBytes * 100) / _totalBytes);
+                                if (percentComplete > 100)
+                                {
+                                    percentComplete = 100;
+                                }
```

Comment:
What is the scenario where this could be needed? Initial underestimation, or files modified mid-copy? Anyways, it can be simplified with `Math.Min`:

```suggestion
                                var percentComplete = (int) Math.Min(_copiedBytes * 100 / _totalBytes, 100);
```

Reply:
If the copying catches up to the thread calculating the totalbytes, there is a race condition where percentComplete is temporarily > 100